manim/utils/space_ops.py

```diff
@@ -53,15 +60,22 @@ def get_norm(vect):
     return np.linalg.norm(vect)
 
 
-def norm_squared(v):
+def norm_squared(v: float) -> np.ndarray:
```

Comment:
This doesn't return an array, it returns a float

Reply:
```suggestion
def norm_squared(v: float) -> float:
```

Reply:
https://numpy.org/doc/stable/reference/generated/numpy.dot.html
It gives back a ndarray

Reply:
In our case the vectors are one-dimensional arrays, the result of the dot product is a number. `float` is fine:

```py
>>> np.dot(np.array([1, 2, 3]), np.array([1.5, 2.5, 3.5]))
17.0
>>> type(_)
<class 'numpy.float64'>
```